### ISSUE TITLE
[Bug] Bad interconnection between other Fediverse instances

### ISSUE THREAD

**Issue by pylapp** [pixelfed/pixelfed#3271]
_**Given** I posted a picture on Pixelfed (given @pylapp@pixelfed.social)
**When** someone on Mastodon (given @anciensEnssat@mamot.fr) "boosted / likes" my publication on its Mastodon client
**Then** the Pixelfed publication shows the user (given @anciensEnssat) liked my publication
**But when** I click on its profile alias (supposed to be @anciensEnssat@mamot.fr on Mastodon)
**Then** I am redirected to Pixelfed page for the user name (@anciensEnssat) and not redirected to Mastodon_

In a nutshell, handles from fediverse are managed always as Pixelfed users and not users from fediverse.
In the best case we are redirected to a 404 page on Pixelfed because the suer does not exists.
In the worst case we are redirected to a Pixelfed suer who can be another unrelated unexpected user.

See more detail on the annotated screenshot bellow.

![annotated screenshot](https://user-images.githubusercontent.com/7559007/155298049-c2e56252-22ef-4339-b212-75a92a8a72a5.jpg)

**Comment by dansup** [comment]
@pylapp This has been fixed in the latest commits! This had nothing to do with federation btw, it was caused by our UI not differentiating between local and remote accounts, it did not affect federation at all.
